src/os/bluestore/ZonedAllocator.cc

```diff
@@ -92,12 +102,31 @@ int64_t ZonedAllocator::allocate(
 		 << " located at zone " << zone_num
 		 << " and zone offset " << offset % zone_size << dendl;
 
+  if (zones_to_clean.empty() && low_on_space()) {
+    find_zones_to_clean(); 
+    cleaner_lock->lock();
```

Comment:
may be move inside find_zones_to_clean?
I'd prefer to have all the synchronization stuff (which includes num_zones_to_clean var now) in a single place

Reply:
Done.